### PR TITLE
feat: add Namespace CLI via Github Releases

### PR DESCRIPTION
**Pull request by pl4nty**
Migrated from https://github.com/devcontainers-contrib/features/pull/591

### CODE REVIEW

File: test/namespace-cli/scenarios.json
Comment (koralowiec) [inline test/namespace-cli/scenarios.json:1]: Similarly to https://github.com/devcontainers-extra/features/pull/55, please add an additional test for a specific version ([comment](https://github.com/devcontainers-extra/features/pull/55/files#r1806929253))

**Review by koralowiec — CHANGES_REQUESTED** [review]
Thanks for the PR!
I left one change request

File: test/namespace-cli/scenarios.json
Comment (pl4nty) [reply test/namespace-cli/scenarios.json:1]: I've added a version test for the CLI itself, but the accompanying docker credential helper doesn't have a version indicator

Comment (koralowiec) [reply test/namespace-cli/scenarios.json:1]: That's fine, if the helper doesn't expose version information, then we can skip it

**Review by koralowiec — APPROVED** [review]

**Comment by koralowiec** [comment]
Let's merge it!

**Pull request merged**